apps/web/src/app/settings/verify-email.component.ts

```diff
@@ -12,25 +13,39 @@ import { PlatformUtilsService } from "@bitwarden/common/abstractions/platformUti
 export class VerifyEmailComponent {
   actionPromise: Promise<unknown>;
 
+  @Output() onVerified = new EventEmitter<boolean>();
+
   constructor(
     private apiService: ApiService,
     private i18nService: I18nService,
     private platformUtilsService: PlatformUtilsService,
-    private logService: LogService
+    private logService: LogService,
+    private tokenService: TokenService
   ) {}
 
+  async verifyEmail(): Promise<void> {
+    await this.apiService.refreshIdentityToken();
+    if (await this.tokenService.getEmailVerified()) {
+      this.onVerified.emit(true);
+      this.platformUtilsService.showToast("success", null, this.i18nService.t("emailVerified"));
+    } else {
+      await this.apiService.postAccountVerifyEmail();
+      this.platformUtilsService.showToast(
+        "success",
+        null,
+        this.i18nService.t("checkInboxForVerification")
+      );
+    }
+  }
```

Comment:
I typically prefer to use guards clauses here. But not a strong preference.

```suggestion
    if (await this.tokenService.getEmailVerified()) {
      this.onVerified.emit(true);
      this.platformUtilsService.showToast("success", null, this.i18nService.t("emailVerified"));
      return;
    }
    
    await this.apiService.postAccountVerifyEmail();
    this.platformUtilsService.showToast(
      "success",
      null,
      this.i18nService.t("checkInboxForVerification")
    );
  }
```

Reply:
I like the suggestion! I don't mind making those changes, feel free to commit the suggestion or I can make them and test it out later tonight (PST time).